src/hooks/useRunOnFirstRender.ts

```diff
@@ -0,0 +1,43 @@
+import {useEffect, useRef} from 'react';
+import useBeforeRemove from './useBeforeRemove';
+
+/**
+ * Hook that runs a callback **only once on the first render** of a component,
+ * and resets its "first render" state when the component is removed from navigation.
+ *
+ * @param firstRenderCb - The callback function to execute on the component's first render.
+ * @param shouldSkip - Optional condition to control whether the callback should run.
+ *   - If `true`, the callback will we skipped.
+ *   - If `false` (default), AND it's a first render, the callback will be executed.
+ *
+ * @example
+ * useRunOnFirstRender(() => {
+ *   console.log("Component mounted for the first time!");
+ * });
+ *
+ * @example
+ * // Run the callback only if condition is true
+ * useRunOnFirstRender(() => {
+ *   console.log("Runs on first render when condition is true");
+ * }, someBooleanCondition);
+ */
+function useRunOnFirstRender(firstRenderCb: () => void, shouldSkip = false) {
```

Comment:
are we sure we want to expose this hook? seems like an antipattern to me

Reply:
This is definitely against react rules but it is true we use this pattern rather often in the app, so I'm not totally against it.

Reply:
Though I would change the name of this function (or split it into two different hooks):
- [useConditionalEffect](https://github.com/streamich/react-use/pull/1882/files/c3ddd44b23375069cde7f22af2387870f4c1a313)
- `useEffectOnce`/`useEffectOnMount` for running an effect on mount (maybe this one is redundant because one could use useEffect with empty dependency list)